### ISSUE TITLE
similarity task run, but didn't do anything

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#939]
https://tools.taskcluster.net/groups/EtQ7qTyZTEOtwTxBsnTA0A/tasks/SLjLm837QmeXAEQaW45-wA/runs/0/logs/public%2Flogs%2Flive.log

The artifact is not available. I think the training didn't happen at all.

@ayush-1506 you can debug this locally, just build and run the docker image.
```
docker build -t mozilla/bugbug-base-nlp -f infra/dockerfile.base-nlp .
docker run -it --rm mozilla/bugbug-base-nlp bash
```
Then in the Docker container, you can run the command from the task (https://github.com/mozilla/bugbug/blob/029d93ada4221642f2bc65b1ce546ec1f0d40fec/infra/data-pipeline.yml#L481-L483):
```
bugbug-train-similarity --algorithm bm25
```

Another issue that we definitely need to solve is that the similarity module is building the `duplicates` map as soon as it is imported, and when it is imported the bugs database is not available yet (since we import at https://github.com/mozilla/bugbug/blob/029d93ada4221642f2bc65b1ce546ec1f0d40fec/scripts/similarity_trainer.py#L13, and download the DB later at https://github.com/mozilla/bugbug/blob/029d93ada4221642f2bc65b1ce546ec1f0d40fec/scripts/similarity_trainer.py#L47-L48).
We can simply move the `duplicates` map building to the `evaluation` function, since that's the only user of the map.

**Comment by marco-c** [comment]
Unfortunately it still happened :(

https://tools.taskcluster.net/groups/FECJFy8CTIy-ZCrAt8jgIA/tasks/fBO2GNGbRC2sSwAVcKkuOQ/runs/0/logs/public%2Flogs%2Flive.log

**Comment by marco-c** [comment]
Fixed by 16e36cb54b45058352dd202daf116feb0f0d63db.
